OpenRA.Mods.Common/Traits/World/PaletteFromPlayerPaletteWithAlpha.cs

```diff
@@ -21,11 +21,13 @@ class PaletteFromPlayerPaletteWithAlphaInfo : ITraitInfo
 	{
 		[FieldLoader.Require]
 		[Desc("The prefix for the resulting player palettes")]
-		public readonly string BaseName = null;
+		[PaletteDefinition(true)] public readonly string BaseName = null;
 
 		[FieldLoader.Require]
 		[Desc("The name of the player palette to base off.")]
-		public readonly string BasePalette = null;
+		[PaletteReference("IsPlayerPalette")] public readonly string BasePalette = null;
```

Comment:
The description will need to be updated to reflect the now-optional player-paletteness.